### PR TITLE
[ID: 501, 502, 505] When and amount paid requirements (partial)

### PR DESCRIPTION
**Pull request by Hjortsberg**
Please have a look at these last tests.

### CODE REVIEW

File: test/PaymentTest.java [hunk 1]
```diff
@@ -97,6 +98,36 @@ public void aboveHalfCompletionRatio() throws IOException {
         amount = payment.getMonthlyAmount("19901201-4509", 0, 100, 50);
         assertTrue(amount > 0);
     }
+
+    @Test
+    public void fullTimeLoan() throws IOException {
+        PaymentImpl payment = new PaymentImpl(new CalendarMock());
+        int amount;
```
Comment (willeponken) [inline test/PaymentTest.java]: It's kind of weird that you initialize the value here, and later sets the content, when you only set it once!

File: test/PaymentTest.java [hunk 2]
```diff
@@ -97,6 +98,36 @@ public void aboveHalfCompletionRatio() throws IOException {
...
+
+        amount = payment.getMonthlyAmount("19901201-4509", 0, 100, 100);
+        assertEquals(FULL_LOAN, amount - FULL_SUBSIDY);
+    }
+
+    @Test
+    public void fullTimeSubsidy() throws IOException {
+        PaymentImpl payment = new PaymentImpl(new CalendarMock());
+        int amount;
```
Comment (willeponken) [inline test/PaymentTest.java]: The same problem here.

File: test/PaymentTest.java [hunk 3]
```diff
@@ -97,6 +98,36 @@ public void aboveHalfCompletionRatio() throws IOException {
...
+
+        amount = payment.getMonthlyAmount("19901201-4509", 0, 100, 100);
+        assertEquals(FULL_SUBSIDY, amount - FULL_LOAN);
+    }
+
+    @Test
+    public void entitledFullAmount() throws IOException {
+        PaymentImpl payment = new PaymentImpl(new CalendarMock());
+        int amount;
```
Comment (willeponken) [inline test/PaymentTest.java]: This looks perfect though! Good job.

File: test/PaymentTest.java (hunk 2 above)
Comment (Hjortsberg) [reply test/PaymentTest.java]: I agree, fixed

**Pull request merged**